src/mpi/coll/transports/mpich/transport.h

```diff
@@ -82,15 +94,10 @@ static inline int MPIC_MPICH_init()
     return 0;
 }
 
-static inline int MPIC_MPICH_comm_init(MPIC_MPICH_comm_t * comm, void *base)
+static inline void MPIC_MPICH_sched_cache_store( MPIC_MPICH_comm_t *comm,
+                                    void* key, int len, void* s)
 {
-    MPIR_Comm *mpi_comm = container_of(base, MPIR_Comm, ch4_coll);
-    comm->mpid_comm = mpi_comm;
-}
-
-static inline int MPIC_MPICH_comm_cleanup(MPIC_MPICH_comm_t * comm)
-{
-    comm->mpid_comm = NULL;
+    MPIC_add_sched( &(comm->sched_cache), key, len, s);
```

Comment:
Since all this function is doing is making this function call, can we remove MPIC_add_sched and move its content  here.